### PR TITLE
chore(ES-86): keep models up-to-date and add target models for all technologies

### PR DESCRIPTION
**Pull request by eliasmueller**
# Changes

Please provide a brief description of the changes here.

## Merge Requirements

For new features contributions, please make sure you have completed the following
essential items:

* [ ] `CHANGELOG.md` updated to document new feature additions
* [ ] Appropriate documentation updates in the [docs][]
* [ ] Appropriate Helm chart updates in the [helm-charts][]

Maintainers will not merge until the above have been completed. If you're unsure
which docs need to be changed ping the
[@open-telemetry/demo-approvers](https://github.com/orgs/open-telemetry/teams/demo-approvers).

[docs]: https://opentelemetry.io/docs/demo/
[helm-charts]: https://github.com/open-telemetry/opentelemetry-helm-charts

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,140 +1,40 @@
...
+Therefore, a [MongoDB](http://mongodb.com) is added, which adds the products of the shop into the database as JSON entries.
+In the default variant of the astronomy shop, only a plain file is used to provide the products.
+
+Thus, the logic of the `productcatalogservice` is adapted, to initialize and use the additionally provided MongoDB.
+
+The changes of the `productcatalogservice` and the addition of the MongoDB is integrated into all deployment variants in this fork.
+
+### Translation to Ansible
+
+Lorem # TODO
```
Comment (eliasmueller) [inline README.md:32]: noch nicht ganz fertig

File: README.md [hunk 2]
```diff
@@ -1,140 +1,64 @@
...
+
+Due to the imperative character of Ansible, in the [Ansible target model](./edmm/otel-store_ansible_translated.yaml), each component has operations representing the installation of the Docker containers.
 
-- Provide a realistic example of a distributed system that can be used to
-  demonstrate OpenTelemetry instrumentation and observability.
-- Build a base for vendors, tooling authors, and others to extend and
-  demonstrate their OpenTelemetry integrations.
-- Create a living example for OpenTelemetry contributors to use for testing new
-  versions of the API, SDK, and other components or enhancements.
+In the [Terraform target model](./edmm/otel-store_terraform_translated.yaml), there are common component types for all docker_containers,
```
Comment (heidrifx) [inline README.md]: Personally, I would replace "docker_container" by either "docker container" or use Markdown code highlighting, i.e. `docker_container`

File: README.md [hunk 3]
```diff
@@ -1,140 +1,64 @@
...
-[Splunk]: https://github.com/signalfx/opentelemetry-demo
-[SumoLogic]: https://www.sumologic.com/blog/common-opentelemetry-demo-application/
-[TelemetryHub]: https://github.com/TelemetryHub/opentelemetry-demo/tree/telemetryhub-backend
-[Teletrace]: https://github.com/teletrace/opentelemetry-demo
-[Tracetest]: https://github.com/kubeshop/opentelemetry-demo
-[Uptrace]: https://github.com/uptrace/uptrace/tree/master/example/opentelemetry-demo
+For the Kubernetes model there are two variants. The [first](./edmm/otel-store_k8s_translated.yaml) is the actual result of the translation, lacking _hosted_on_ relations.
+That is, as the Kubernetes plugin only creates hosted_on relations when a component named "container_runtime" with a matching type is present in the TADM.
+Therefore, there is a [second target model](./edmm/otel-store_k8s_translated_with_container_runtime.yaml) where this "container_runtime" has been manually added, before creating the relations.
+Thus, all components obtain a _hosted_on_ relation.
```
Comment (heidrifx) [inline README.md]: - "hosted_on" was called "hosted-on" earlier, why change it here?
- same remark on "container_runtime" as with the "docker_container" in the other comment

File: terraform/README.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+### TERRAFORM
+To execute the terraform build run:
+```shell
+terraform init && terraform apply
+```
```
Comment (heidrifx) [inline terraform/README.md]: Correct me if I'm wrong but the `&&` will not work under Windows' CMD/Powershell
So maybe add a separate command hint for Windows?

File: README.md [hunk 5]
```diff
@@ -1,140 +1,64 @@
...
+
+The changes of the `productcatalogservice` and the addition of the MongoDB is integrated into all deployment variants in this fork.
+
+### Translation to Ansible
+
+To verify the [Ansible MPS Plugin](https://github.com/UST-DeMAF/ansible-mps-plugin) the Astronomy Shop is available as Ansible installation.
+It is based on the [Docker-Compose](https://opentelemetry.io/docs/demo/docker_deployment/) and installs all components of the shop as docker containers on the local Docker runtime.
+For each component there is a dedicated Ansible Role representing the loose coupling of the components.
+
+A usage guide can be found in the dedicated [README in /ansible](./ansible/README.md).
```
Comment (heidrifx) [inline README.md:36]: The README only has a command which will work on UNIX-like systems (i.e. MacOS and Linux) but calling `sudo` will most likely fail on Windows. Can we extend the README by a command which will work for Windows?

File: .github/workflows/build-productcatalogservice-fork.yml
Comment (heidrifx) [inline .github/workflows/build-productcatalogservice-fork.yml:1]: I have multiple remarks on this 😅:
- Why not use the workflow from our plugins as template?
  - i.e. QEMU is missing, action versions are not up-to-date
- Running the workflow on every push to main _and_ demaf will build images with the same tag thus overwriting the old one. So for example if the main branch works fine and someone pushes an incomplete version to demaf, the working docker image will _not_ be the latest one available.
  - In my opinion we should also use a _testing_ or _dev_ tag.
- Who is the `github.repository_owner` of this fork? Someone from the UST-DeMAF organization? If it is the original repository's owner maybe we should hardcode some one from our team?
- Did the arm64 version work locally?
- I find the usage of the `IMAGE_NAME` as suffix to the `IMAGE_TAG` kind of confusing. Why not simply extend the `IMAGE_TAG`?
- Lastly, I would change the image name from "demo" to "opentelemetry-demo" or "otel-demo". "demo" alone is a little confusing in my opinion as we also have other demos like the Kube example.

File: kubernetes/opentelemetry-demo.yaml [hunk 6]
```diff
@@ -10461,7 +10461,7 @@ spec:
       serviceAccountName: opentelemetry-demo
       containers:
         - name: productcatalogservice
-          image: 'ghcr.io/open-telemetry/demo:1.11.1-productcatalogservice'
+          image: 'ghcr.io/ust-demaf/demo:1.11.1-productcatalogservice'
```
Comment (heidrifx) [inline kubernetes/opentelemetry-demo.yaml:10464]: _If_ the workflow is changed this needs to be updated as well!

**Review by heidrifx — CHANGES_REQUESTED** [review]
I'll mark this as _"request changes"_ to not accidentally push a docker image to our ghcr if we choose to change the naming

File: .github/workflows/build-productcatalogservice-fork.yml
Comment (eliasmueller) [reply .github/workflows/build-productcatalogservice-fork.yml:1]: Regarding the naming and tagging; I chose to use the exact same naming strategy as in the original repository, to be consistent, even though I agree it is kinda confusing;

Regarding the arm64, yes.

Regarding the owner, it is Marcel as it is forked into UST-DeMAF

Regarding the workflow -> yes I can adapt it to match the others

Comment (eliasmueller) [reply .github/workflows/build-productcatalogservice-fork.yml:1]: adjusted the workflow to match the others

File: .github/workflows/buildAndPushImage-productcatalogservice.yml [hunk 7]
```diff
@@ -14,24 +13,29 @@ env:
...
   build-using-dockerfile-push-2-ghcr:
     runs-on: ubuntu-latest
-
     steps:
-      # Checkout the repository
-      - name: Checkout repository
-        uses: actions/checkout@v3
+      - uses: actions/checkout@v4
+        with:
+          submodules: 'true'
```
Comment (heidrifx) [inline .github/workflows/buildAndPushImage-productcatalogservice.yml]: we are not using submodules for this, are we?
this does not hurt in any way but might be confusing

**Review by heidrifx — COMMENTED** [review]
lgtm but please take a last small look at the comment 😅

**Pull request merged**